dlc2nwb/utils.py

```diff
@@ -123,37 +98,105 @@ def convert_h5_to_nwb(config, h5file, individual_name="ind1"):
         )  # setting timestamps to dummy TODO: extract timestamps in DLC?
     else:
         timestamps = get_movie_timestamps(video[0])
+    return scorer, df, video, paf_graph, timestamps, cfg
+
+
+def _write_pes_to_nwbfile(nwbfile, animal, df_animal, scorer, video, paf_graph, timestamps):
+    pose_estimation_series = []
+    for kpt, xyp in df_animal.groupby(level="bodyparts", axis=1, sort=False):
+        data = xyp.to_numpy()
+
+        pes = PoseEstimationSeries(
+            name=f"{animal}_{kpt}",
+            description=f"Keypoint {kpt} from individual {animal}.",
+            data=data[:, :2],
+            unit="pixels",
+            reference_frame="(0,0) corresponds to the bottom left corner of the video.",
+            timestamps=timestamps,
+            confidence=data[:, 2],
+            confidence_definition="Softmax output of the deep neural network.",
+        )
+        pose_estimation_series.append(pes)
+
+    pe = PoseEstimation(
+        pose_estimation_series=pose_estimation_series,
+        description="2D keypoint coordinates estimated using DeepLabCut.",
+        original_videos=[video[0]],
+        # TODO check if this is a mandatory arg in ndx-pose (can skip if video is not found_
+        dimensions=[list(map(int, video[1].split(",")))[1::2]],
+        scorer=scorer,
+        source_software="DeepLabCut",
+        source_software_version=__version__,
+        nodes=[pes.name for pes in pose_estimation_series],
+        edges=paf_graph,
+    )
+    if 'behavior' in nwbfile.processing:
+        behavior_pm = nwbfile.processing["behavior"]
+    else:
+        behavior_pm = nwbfile.create_processing_module(
+            name="behavior", description="processed behavioral data"
+        )
+    behavior_pm.add(pe)
+    return nwbfile
```

Comment:
this function writes the pose estimation series of a given animal to a specific nwbfile (already created in memory)